syntax = "proto3";

service UserService {
    rpc Authentication (AuthRequest) returns (AuthResponse);
    rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
    rpc GetUserById (GetUserByIdRequest) returns (GetUserByIdResponse);
}

message CreateUserRequest {
    string name = 1;
    string email = 2;
    string password = 3;
}

message CreateUserResponse {
    bool isSuccess = 1;
    int32 userId = 2;
    string errorMsg = 3;
}

message AuthRequest {
    string email = 1;
    string password = 2;
}

message AuthResponse {
    bool isSuccess = 1;
    string errorMsg = 2;
    string accessToken = 3;
    string refreshToken = 4;
}

message GetUserByIdRequest {
    int32 id = 1;
}

message GetUserByIdResponse {
    bool isSuccess = 1;
    string errorMsg = 2;
    string name = 3;
}
